applications/StructuralMechanicsApplication/custom_constitutive/plasticity_isotropic_kinematic_j2.cpp

```diff
@@ -0,0 +1,525 @@
+    }
+    return false;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+bool& PlasticityIsotropicKinematicJ2::GetValue(
+    const Variable<bool>& rThisVariable,
+    bool& rValue
+    )
+{
+    if(rThisVariable == INELASTIC_FLAG){
+        rValue = mPlasticEvolution;
+    }
+
+    return rValue;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::InitializeMaterial(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues
+    )
+{
+    mPlasticStrainLast = ZeroVector(this->GetStrainSize());
+    mPlasticStrain = ZeroVector(this->GetStrainSize());
+    mEquivalentPlasticStrainLast = 0.0;
+    mEquivalentPlasticStrain = 0.0;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::FinalizeSolutionStep(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    mPlasticStrainLast = mPlasticStrain;
+    mEquivalentPlasticStrainLast = mEquivalentPlasticStrain;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::InitializeMaterialResponsePK1(ConstitutiveLaw::Parameters& rValues)
+{
+    InitializeMaterialResponseCauchy(rValues); //not yet implemented
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::InitializeMaterialResponsePK2(ConstitutiveLaw::Parameters& rValues)
+{
+    InitializeMaterialResponseCauchy(rValues); //not yet implemented
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::InitializeMaterialResponseKirchhoff(ConstitutiveLaw::Parameters& rValues)
+{
+    InitializeMaterialResponseCauchy(rValues); //not yet implemented
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::InitializeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
+{
+    mPlasticStrainLast = ZeroVector(this->GetStrainSize());
```

Comment:
what is the PLasticStrainLast and PlasticStrain? please do not store intermediate values

Reply:
ok. I will store only one (last converged strain)